src/background.js

```diff
@@ -109,46 +110,61 @@ function scheduleReload() {
 		return
 	}
 
-	const now = new Date()
-	let start
+	const mode = store.get('settings.autoReloadMode')
 
-	const reloadHour = store.get('settings.autoReloadHour') || 0
+	if (mode === 'hour') {
+		const now = new Date()
+		let start
 
-	if (now.getHours() < reloadHour) {
-		start = new Date(
-			now.getFullYear(),
-			now.getMonth(),
-			now.getDate(),
-			reloadHour,
-			0,
-			0,
-			0
-		)
-	} else {
-		start = new Date(
-			now.getFullYear(),
-			now.getMonth(),
-			now.getDate() + 1,
-			reloadHour,
-			0,
-			0,
-			0
-		)
-	}
+		const reloadHour = store.get('settings.autoReloadHour') || 0
 
-	const wait = start.getTime() - now.getTime()
+		if (now.getHours() < reloadHour) {
+			start = new Date(
+				now.getFullYear(),
+				now.getMonth(),
+				now.getDate(),
+				reloadHour,
+				0,
+				0,
+				0
+			)
+		} else {
+			start = new Date(
+				now.getFullYear(),
+				now.getMonth(),
+				now.getDate() + 1,
+				reloadHour,
+				0,
+				0,
+				0
+			)
+		}
 
-	reloadTimeout = setTimeout(
-		() => {
+		const wait = start.getTime() - now.getTime()
+
+		reloadTimeout = setTimeout(
+			() => {
+				if (reloadTimeout) {
+					clearTimeout(reloadTimeout)
+					reloadTimeout = null
+				}
+				win.reload()
+				scheduleReload()
+			},
+			wait < 0 ? 0 : wait
+		)
+	} else if (mode === 'every') {
+		const reloadEvery = store.get('settings.autoReloadEvery') || '1h30m'
+		const ms = parse(reloadEvery)
```

Comment:
Does this work if someone inputs a number in milliseconds?

Reply:
yes